app/src/main/java/db/TimelogEntryModel.java

```diff
@@ -0,0 +1,138 @@
+import android.util.Log;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.util.Date;
+
+public class TimelogEntryModel {
+  private static ContentResolver contentResolver;
+  private Uri uri;
+  private long id;
+  private Date startDate;
+  private Date endDate;
+  private String comment;
+
+  private DateFormat _UI_DATE_FORMATTER = DateFormat.getDateTimeInstance(
+      DateFormat.SHORT,
+      DateFormat.SHORT
+  );
+
+  public static void setContentResolver( ContentResolver resolver ) {
+    TimelogEntryModel.contentResolver = resolver;
+  }
+
+  public TimelogEntryModel( Date startDate, Date endDate, String comment ) {
+    Log.d( "Model", "Value constructor called");
+    this.id = -1;
+    this.uri = null;
+    this.startDate = startDate;
+    this.endDate = endDate;
+    this.comment = comment;
+  }
+
+  private TimelogEntryModel() {}
+
+  // Assumes cursor is not null and points to the correct provider record
+  public TimelogEntryModel( final Cursor cursor ) throws ParseException {
+    Log.d( "Model", "Cursor constructor called");
+    final int idColumnIndex = cursor.getColumnIndex( BaseColumns._ID );
+    final int startDateColumnIndex = cursor.getColumnIndex( TimelogContract.Timelog.Columns.START );
+    final int endDateColumnIndex = cursor.getColumnIndex( TimelogContract.Timelog.Columns.END );
+    final int commentColumnIndex = cursor.getColumnIndex( TimelogContract.Timelog.Columns.COMMENT );
+
+    this.id = cursor.getLong( idColumnIndex );
+    this.uri = ContentUris.withAppendedId( TimelogContract.Timelog.CONTENT_URI, this.id );
+    this.startDate = parseDbDate( cursor.getString( startDateColumnIndex ) );
+    this.endDate = parseDbDate( cursor.getString( endDateColumnIndex ) );
+    this.comment = cursor.getString( commentColumnIndex );
+  }
+
+  public boolean save() {
+    return ( this.uri == null || this.id == -1 ) ? insert() : update();
+  }
+
+  protected boolean insert() {
+    ContentValues values = new ContentValues();
+    values.put( TimelogContract.Timelog.Columns.START, getDbStartDateString() );
+    values.put( TimelogContract.Timelog.Columns.END, getDbEndDateString() );
+    values.put( TimelogContract.Timelog.Columns.COMMENT, this.comment );
+
+    this.uri = TimelogEntryModel.contentResolver.insert( TimelogContract.Timelog.CONTENT_URI, values );
+    this.id = ContentUris.parseId( this.uri );
+
+    // FIXME: Error handling; how do we detect an insert failed?
+    // TODO: return false on error
+    return true;
+  }
+
+  protected boolean update() {
+    ContentValues values = new ContentValues();
+    values.put( TimelogContract.Timelog.Columns.START, getDbStartDateString() );
+    values.put( TimelogContract.Timelog.Columns.END, getDbEndDateString() );
+    values.put( TimelogContract.Timelog.Columns.COMMENT, this.comment );
+
+    return ( TimelogEntryModel.contentResolver.update( this.uri, values, null, null) == 1 );
+  }
+
+  public String getUiStartDateString() {
```

Comment:
Wie gesagt: Eigentlich gehört sowas in einen Presenter, aber da momentan nur zwei Methoden hier gebraucht werden, hab ich mir die Presenter-Klasse erstmal gespart. Sicher: Sauber ist anders.